services/director-v2/src/simcore_service_director_v2/modules/db/repositories/projects.py

```diff
@@ -22,3 +23,12 @@ async def get_project(self, project_id: ProjectID) -> ProjectAtDB:
         if not row:
             raise ProjectNotFoundError(project_id)
         return ProjectAtDB.from_orm(row)
+
+    async def is_node_present_in_workbench(
+        self, project_id: ProjectID, node_uuid: NodeID
+    ) -> bool:
+        try:
+            project = await self.get_project(project_id)
+            return f"{node_uuid}" in project.workbench
```

Comment:
It does the work but I believe this check can be directly checked on the postgres server.